conf/schemas/annotations.schema.js

```diff
@@ -0,0 +1,84 @@
+db.runCommand({
+  collMod: "annotations",
+  validator: {
+    $and: [
+      {
+        $or: [{ _user: { $type: "objectId" } }, { _user: { $exists: false } }],
+      },
+      {
+        _content: { $type: "object", $exists: true },
+      },
+      {
+        "_content.contentType": { $type: "string", $exists: true },
+      },
+      {
+        "_content._id": { $type: "string", $exists: true },
+      },
+      {
+        $or: [{ _task: { $type: "objectId" } }, { _task: { $exists: false } }],
+      },
+      {
+        team: { $type: "string", $exists: true },
+      },
+      {
+        state: { $type: "object", $exists: true },
+      },
+      {
+        "state.isAssigned": { $type: "bool", $exists: true },
+      },
+      {
+        "state.isFinished": { $type: "bool", $exists: true },
+      },
+      {
+        "state.isInProgress": { $type: "bool", $exists: true },
+      },
+      {
+        typ: {
+          $exists: true,
+          $in: [
+            "Task",
+            "View",
+            "Explorational",
+            "CompoundTask",
+            "CompoundProject",
+            "CompoundTaskType",
+            "Tracing Base",
+            "Orphan",
+          ],
+        },
+      },
+      {
+        version: { $type: "number", $exists: true },
+      },
+      {
+        $or: [{ _name: { $type: "string" } }, { _name: { $exists: false } }],
+      },
+      {
+        $or: [{ tracingTime: { $type: "number" } }, { tracingTime: { $exists: false } }],
+      },
+      {
+        $or: [{ created: { $type: "number" } }, { created: { $exists: false } }],
+      },
+      {
+        _id: { $type: "objectId", $exists: true },
+      },
+      {
+        isActive: { $type: "bool", $exists: true },
+      },
+      {
+        $or: [{ readOnly: { $type: "bool" } }, { readOnly: { $exists: false } }],
+      },
+      {
+        isPublic: { $type: "bool", $exists: true },
+      },
+      {
+        tags: { $exists: true },
```

Comment:
This is mongo specific because the $type matches against the elements of an array. (https://docs.mongodb.com/manual/reference/operator/query/type/#arrays)

Reply:
Great, another reason to love mongo :/ So there is no way to tell `{ tags: [ "a", "b", "c" ] }` and `{ tags: "a" }` apart, since they are both `$type: "string"`?
I would still suggest adding a `$type: "string"` for the tags then.

Reply:
Yeah that's true :/
Did you see the `$or` below?
I think I splitted this because the tags array is required but it might be empty, so the `$type` check would fail in that case.